Sources/MessageInputBar.swift

```diff
@@ -24,133 +24,434 @@
+    }()
+    
+    open let bottomStackView: UIStackView = {
+        let view = UIStackView()
+        view.axis = .horizontal
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.distribution = .fill
+        view.alignment = .fill
+        view.spacing = 15
+        return view
+    }()
+    
+    open lazy var textView: InputTextView = { [unowned self] in
+        let textView = InputTextView()
+        textView.translatesAutoresizingMaskIntoConstraints = false
+        textView.inputBarAccessoryView = self
+        return textView
+        }()
+    
+    /// The padding around the textView that separates it from the stackViews
+    open var textViewPadding: UIEdgeInsets = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0) {
+        didSet {
+            textViewLayoutSet?.bottom?.constant = -textViewPadding.bottom
+            textViewLayoutSet?.left?.constant = textViewPadding.left
+            textViewLayoutSet?.right?.constant = -textViewPadding.right
+            bottomStackViewLayoutSet?.top?.constant = textViewPadding.bottom
+        }
     }
-
-    convenience public init() {
+    
+    open lazy var sendButton: InputBarButtonItem = { [unowned self] in
+        return InputBarButtonItem()
+            .configure {
+                $0.size = CGSize(width: 52, height: 36)
+                $0.spacing = .fixed(4)
+                $0.isEnabled = false
+                $0.title = "Send"
+                $0.titleLabel?.font = UIFont.preferredFont(forTextStyle: .headline)
+            }.onTextViewDidChange({ (item, textView) in
+                item.isEnabled = !textView.text.isEmpty
+            }).onTouchUpInside {
+                $0.inputBarAccessoryView?.didSelectSendButton()
+        }
+        }()
+    
+    /// The anchor contants used by the UIStackViews and InputTextView to create padding within the InputBarAccessoryView
+    open var padding: UIEdgeInsets = UIEdgeInsets(top: 6, left: 12, bottom: 6, right: 12) {
+        didSet {
+            updateViewContraints()
+        }
+    }
+    
+    override open var intrinsicContentSize: CGSize {
+        let sizeToFit = textView.sizeThatFits(CGSize(width: textView.bounds.width, height: .greatestFiniteMagnitude))
+        var heightToFit = sizeToFit.height.rounded() + padding.top + padding.bottom
+        if heightToFit >= maxHeight {
+            textView.isScrollEnabled = true
+            heightToFit = maxHeight
+        } else {
+            textView.isScrollEnabled = false
+        }
+        let size = CGSize(width: bounds.width, height: heightToFit)
+        if previousIntrinsicContentSize != size {
+            delegate?.inputBar?(self, didChangeIntrinsicContentTo: size)
+        }
+        previousIntrinsicContentSize = size
+        return size
+    }
+    
+    /// The maximum intrinsicContentSize height. When reached the delegate 'didChangeIntrinsicContentTo' will be called.
+    open var maxHeight: CGFloat = 288 {
+        didSet {
```

Comment:
Not too worried about this now but it would be nicer if the default value took into account the size of the device's screen. I'm not sure how this would affect the iPad.

Reply:
How about `open var maxHeight: CGFloat = UIScreen.main.bounds.height / {some constant}`